app/services/account_builder.rb

```diff
@@ -0,0 +1,93 @@
+# Builds an Account object given params.
+# This class supports extensions from engines via the `build_subclass` method.
+class AccountBuilder
```

Comment:
I started off with instance variables and started passing them all over in the name of making testing easier.  But I'm fine with using the instance variables again.